mlrun/datastore/google_cloud_storage.py

```diff
@@ -23,30 +24,54 @@
 
 # Google storage objects will be represented with the following URL: gcs://<bucket name>/<path> or gs://...
 
+WORKERS = 8
+CHUNK_SIZE = 32 * 1024 * 1024
+
 
 class GoogleCloudStorageStore(DataStore):
     using_bucket = True
 
     def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
+        self._storage_client = None
+        self._transfer_manager = None
+        self.workers = WORKERS
+        self.chunk_size = CHUNK_SIZE
         super().__init__(parent, name, schema, endpoint, secrets=secrets)
 
     @property
-    def filesystem(self):
-        """return fsspec file system object, if supported"""
-        if self._filesystem:
-            return self._filesystem
+    def storage_client(self):
+        if not self._filesystem or self._storage_client:
+            self.create_filesystem_and_client()
+        return self._storage_client
+
+    def create_filesystem_and_client(self):
         try:
             import gcsfs  # noqa
         except ImportError as exc:
             raise ImportError(
                 "Google gcsfs not installed, run pip install gcsfs"
             ) from exc
+        from google.cloud.storage import Client, transfer_manager
+
+        # Creates with the same storage options to avoid credential differences.
+        storage_options = self.get_storage_options()
+        # in order to support az and wasbs kinds.
```

Comment:
What do az and wasbs have to do with gcs?
```suggestion
        # in order to support az and wasbs kinds
```